superset/assets/javascripts/explore/components/ChartContainer.jsx

```diff
@@ -145,6 +146,25 @@ class ChartContainer extends React.PureComponent {
     };
   }
 
+  isAltered() {
+    // Returns all properties that differ in the
+    // current form data and the base form data
+    const fd = this.props.formData || {};
+    const bfd = (this.props.slice && this.props.slice.form_data) || {};
```

Comment:
Here you may want to use the `currentFormData` as the driver of which keys should exist. `slice.form_data` may have unused / deprecated keys that don't affect how the slice is rendered...